src/main/java/com/serotonin/bacnet4j/npdu/ip/IpNetworkUtils.java

```diff
@@ -158,16 +158,60 @@ public static Address toAddress(final int networkNumber, final InetSocketAddress
         return toAddress(networkNumber, addr.getAddress().getAddress(), addr.getPort());
     }
 
-    public static List<InterfaceAddress> getLocalInterfaceAddresses() {
+    
+    /**
+     * Following details of the NIC (Network Interface Card) are stored in this record:
+     * <ul>
+     * <li> Integer format of index value of the NIC assigned by the OS </li>
+     * <li> String format of the name of the NIC </li>
+     * <li> String format of the IP4 Address assigned to the NIC </li>
+     * <li> String format of the the broadcast Address of the NIC </li>
+     * <li> String format of the SubnetMask of the NIC </li>
+     * <li> Integer format of the Network Prefix Length of the NIC </li> 
+     * </ul>
+     */
+    public static record InterfaceInfo(int index, 
+        String interfaceName, 
+        String localIPAddress, 
+        String broadcastAddress, 
+        String subnetMask, 
+        int networkPrefixLength) {}
+
+
+    /**
+     * Convenient method to get the deails of Network Interface card
+     * <p>
+     * Values of the following parameters of each NIC are stored in record format:
+     * <ul>
+     * <li> index (int) - NIC Card Index assigned by the OS
+     * <li> Name of the NIC (String)
+     * <li> IP4 Address assigned to the NIC (String)
+     * <li> Broadcast Address of the NIC (String)
+     * <li> SubnetMask of the NIC calculated based on the NetworkPrefixLength (String)
+     * <li> Network Prefix Length of the NIC (int)
+     * </ul>
+     * @return the list of records of all NIC available in this device
+    */
```

Comment:
A lot of these tags are not closed. Your IDE should be configured to warn about this.